### PR TITLE
97 - Add compatibility with Safari for webcomponents.js

### LINKED ISSUE #97: Component - Safari compatibliity

### PR DESCRIPTION
**Pull request by chikeichan**
this resolves #97

### CODE REVIEW

File: src/component/component.js [hunk 1]
```diff
@@ -47,19 +47,10 @@ Component.prototype.getData = function(dataKey) {
...
-    } else {
-        html = '<' + this.tagName + '></' + this.tagName + '>';
-    }
-
-    temp.innerHTML = html;
-    return temp.querySelector(this.tagName);
+Component.prototype.createElement = function(data) {
+    var component = document.createElement(this.tagName);
+    component.patch(data);
+    return component;
```
Comment (chikeichan) [inline src/component/component.js]: will be refactor when renderHTML is done

File: src/component/component.js [hunk 2]
```diff
@@ -124,6 +119,19 @@ Component.prototype.registerElement = function(opts) {
...
+        function replaceStyleHost(root) {
+            if (window && window.ShadowDOMPolyfill) {
+                var list = root.getElementsByTagName('style');
+                Array.prototype.forEach.call(list, function (style) {
+                    var text = style.textContent
+                        .replace(/:host\b/gm, this.tagName.toLowerCase())
+                        .replace(/::shadow\b/gm, ' ')
+                        .replace(/::content\b/gm, ' ');
+                    style.textContent = text;
+                }.bind(this));
```
Comment (chikeichan) [inline src/component/component.js]: When styling webcomponent, Chrome expect the selector ':host', while Safari expect the tag name itself.

This method use regex to replace :host with tagName when using Polyfill

File: src/component/component.js (hunk 1 above)
Comment (chikeichan) [inline src/component/component.js]: Need to use createElement so that Safari knows to invoke createdCallback

**Comment by sdtsui** [comment]
Notes:

1. createCallback - createdCallback - line 64 of old diff


2. Safari missing :host. Must use tag name.
<custom-element>
  <style>:host{display:block}</style>
</custom-element>


for chrome, :host reference custom-element
the 'host' that is hosting the style tag
yes

**Pull request merged**